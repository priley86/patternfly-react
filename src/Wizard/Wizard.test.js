import React from 'react'
import renderer from 'react-test-renderer'
import { Row, Col } from 'react-bootstrap'

import { Button, Wizard } from '../index'

import {
  mockWizardItems,
  mockLoadingContents
} from './__mocks__/mockWizardItems'

import {
  renderWizardSteps,
  renderSidebarItems,
  renderWizardContents
} from './__mocks__/mockWizardRenderers'

test('Wizard loading renders properly', () => {
  const component = renderer.create(
    <Row>
      <Col sm={12}>
        <Wizard
          className="wizard-pf"
          render={() => {
            return (
              <div>
                <Wizard.Header title="Wizard Title" />
                <Wizard.Body>
                  <Wizard.Row>
                    <Wizard.Main>{mockLoadingContents()}</Wizard.Main>
                  </Wizard.Row>
                </Wizard.Body>
                <Wizard.Footer>
                  <Button bsStyle="default" className="btn-cancel">
                    Cancel
                  </Button>
                  <Button bsStyle="default" disabled>
                    <span className="i fa fa-angle-left" />Back
                  </Button>
                  <Button bsStyle="primary" disabled>
                    Next<span className="i fa fa-angle-right" />
                  </Button>
                </Wizard.Footer>
              </div>
            )
          }}
        />
      </Col>
    </Row>
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Wizard embedded renders properly', () => {
  const component = renderer.create(
    <Row>
      <Col sm={12}>
        <Wizard
          className="wizard-pf"
          steps={mockWizardItems}
          initialStepIndex={0}
          initialSubStepIndex={0}
          render={(
            { activeStepIndex, activeSubStepIndex },
            onStepClick,
            onSidebarItemClick,
            onBackButtonClick,
            onNextButtonClick
          ) => {
            return (
              <div>
                <Wizard.Header title="Wizard Title" />
                <Wizard.Body>
                  <Wizard.Steps
                    steps={renderWizardSteps(
                      mockWizardItems,
                      activeStepIndex,
                      activeSubStepIndex,
                      onStepClick
                    )}
                  />
                  <Wizard.Row>
                    <Wizard.Sidebar
                      items={renderSidebarItems(
                        mockWizardItems,
                        activeStepIndex,
                        activeSubStepIndex,
                        onSidebarItemClick
                      )}
                    />
                    <Wizard.Main>
                      {renderWizardContents(
                        mockWizardItems,
                        activeStepIndex,
                        activeSubStepIndex
                      )}
                    </Wizard.Main>
                  </Wizard.Row>
                </Wizard.Body>
                <Wizard.Footer>
                  <Button bsStyle="default" className="btn-cancel">
                    Cancel
                  </Button>
                  <Button
                    bsStyle="default"
                    disabled={activeStepIndex === 0 && activeSubStepIndex === 0}
                    onClick={onBackButtonClick}
                  >
                    <span className="i fa fa-angle-left" />Back
                  </Button>
                  {(activeStepIndex === 0 || activeStepIndex === 1) && (
                    <Button bsStyle="primary" onClick={onNextButtonClick}>
                      Next<span className="i fa fa-angle-right" />
                    </Button>
                  )}
                  {activeStepIndex === 2 &&
                    activeSubStepIndex === 0 && (
                      <Button bsStyle="primary" onClick={onNextButtonClick}>
                        Deploy<span className="i fa fa-angle-right" />
                      </Button>
                    )}
                  {activeStepIndex === 2 &&
                    activeSubStepIndex === 1 && (
                      <Button bsStyle="primary">
                        Done<span className="i fa fa-angle-right" />
                      </Button>
                    )}
                </Wizard.Footer>
              </div>
            )
          }}
        />
      </Col>
    </Row>
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

/**
 * react-overlays / Jest opened issue
 * Open issue: https://github.com/react-bootstrap/react-overlays/issues/225
 */
// test('Wizard modal renders properly', () => {
//   const showModal = true
//   const handleClose = jest.fn()

//   const component = renderer.create(
//     <div>
//       <Modal
//         show={showModal}
//         onHide={handleClose}
//         dialogClassName="modal-lg wizard-pf"
//       >
//         <Wizard
//           steps={mockWizardItems}
//           initialStepIndex={0}
//           initialSubStepIndex={0}
//           render={(
//             { activeStepIndex, activeSubStepIndex },
//             onStepClick,
//             onSidebarItemClick,
//             onBackButtonClick,
//             onNextButtonClick
//           ) => {
//             return (
//               <div>
//                 <Modal.Header>
//                   <button
//                     className="close"
//                     onClick={handleClose}
//                     aria-hidden="true"
//                     aria-label="Close"
//                   >
//                     <span className="pficon pficon-close" />
//                   </button>
//                   <Modal.Title>Wizard Title</Modal.Title>
//                 </Modal.Header>
//                 <Modal.Body className="wizard-pf-body clearfix">
//                   <Wizard.Steps
//                     steps={renderWizardSteps(
//                       mockWizardItems,
//                       activeStepIndex,
//                       activeSubStepIndex,
//                       onStepClick
//                     )}
//                   />
//                   <Wizard.Row>
//                     <Wizard.Sidebar
//                       items={renderSidebarItems(
//                         mockWizardItems,
//                         activeStepIndex,
//                         activeSubStepIndex,
//                         onSidebarItemClick
//                       )}
//                     />
//                     <Wizard.Main>
//                       {renderWizardContents(
//                         mockWizardItems,
//                         activeStepIndex,
//                         activeSubStepIndex
//                       )}
//                     </Wizard.Main>
//                   </Wizard.Row>
//                 </Modal.Body>
//                 <Modal.Footer className="wizard-pf-footer">
//                   <Button
//                     bsStyle="default"
//                     className="btn-cancel"
//                     onClick={handleClose}
//                   >
//                     Cancel
//                   </Button>
//                   <Button
//                     bsStyle="default"
//                     disabled={activeStepIndex === 0 && activeSubStepIndex === 0}
//                     onClick={onBackButtonClick}
//                   >
//                     <span className="i fa fa-angle-left" />Back
//                   </Button>
//                   {(activeStepIndex === 0 || activeStepIndex === 1) && (
//                     <Button bsStyle="primary" onClick={onNextButtonClick}>
//                       Next<span className="i fa fa-angle-right" />
//                     </Button>
//                   )}
//                   {activeStepIndex === 2 &&
//                     activeSubStepIndex === 0 && (
//                       <Button bsStyle="primary" onClick={onNextButtonClick}>
//                         Deploy<span className="i fa fa-angle-right" />
//                       </Button>
//                     )}
//                   {activeStepIndex === 2 &&
//                     activeSubStepIndex === 1 && (
//                       <Button bsStyle="primary" onClick={handleClose}>
//                         Close<span className="i fa fa-angle-right" />
//                       </Button>
//                     )}
//                 </Modal.Footer>
//               </div>
//             )
//           }}
//         />
//       </Modal>
//     </div>
//   )

//   const tree = component.toJSON()
//   expect(tree).toMatchSnapshot()
// })
