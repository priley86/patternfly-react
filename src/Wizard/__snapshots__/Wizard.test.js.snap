// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wizard embedded renders properly 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-12"
  >
    <div
      className="wizard-pf"
      steps={
        Array [
          Object {
            "label": "1",
            "step": "1",
            "subSteps": Array [
              Object {
                "contents": Object {
                  "label1": "Name",
                  "label2": "Description",
                },
                "label": "1A.",
                "subStep": "1.1",
                "title": "Details",
              },
              Object {
                "contents": Object {
                  "label1": "Lorem ipsum",
                  "label2": "Dolor",
                },
                "label": "1B.",
                "subStep": "1.2",
                "title": "Settings",
              },
            ],
            "title": "First Step",
          },
          Object {
            "label": "2",
            "step": "2",
            "subSteps": Array [
              Object {
                "contents": Object {
                  "label1": "Aliquam",
                  "label2": "Fermentum",
                },
                "label": "2A.",
                "subStep": "2.1",
                "title": "Details",
              },
              Object {
                "contents": Object {
                  "label1": "Consectetur",
                  "label2": "Adipiscing",
                },
                "label": "2B.",
                "subStep": "2.2",
                "title": "Settings",
              },
            ],
            "title": "Second Step",
          },
          Object {
            "label": "3",
            "step": "3",
            "subSteps": Array [
              Object {
                "label": "3A.",
                "subStep": "3.1",
                "title": "Summary",
              },
              Object {
                "label": "3B.",
                "subStep": "3.2",
                "title": "Progress",
              },
            ],
            "title": "Review",
          },
        ]
      }
    >
      <div>
        <div
          className="wizard-pf-header"
        >
          <h4
            className="wizard-pf-title"
          >
            Wizard Title
          </h4>
        </div>
        <div
          className="wizard-pf-body"
        >
          <div
            className="wizard-pf-steps"
          >
            <ul
              className="wizard-pf-steps-indicator"
            >
              <li
                className="wizard-pf-step active"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <span
                    className="wizard-pf-step-number"
                  >
                    1
                  </span>
                  <span
                    className="wizard-pf-step-title"
                  >
                    First Step
                  </span>
                  <span
                    className="wizard-pf-step-title-substep active"
                  >
                    Details
                  </span>
                  <span
                    className="wizard-pf-step-title-substep"
                  >
                    Settings
                  </span>
                </a>
              </li>
              <li
                className="wizard-pf-step"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <span
                    className="wizard-pf-step-number"
                  >
                    2
                  </span>
                  <span
                    className="wizard-pf-step-title"
                  >
                    Second Step
                  </span>
                  <span
                    className="wizard-pf-step-title-substep"
                  >
                    Details
                  </span>
                  <span
                    className="wizard-pf-step-title-substep"
                  >
                    Settings
                  </span>
                </a>
              </li>
              <li
                className="wizard-pf-step"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <span
                    className="wizard-pf-step-number"
                  >
                    3
                  </span>
                  <span
                    className="wizard-pf-step-title"
                  >
                    Review
                  </span>
                  <span
                    className="wizard-pf-step-title-substep"
                  >
                    Summary
                  </span>
                  <span
                    className="wizard-pf-step-title-substep"
                  >
                    Progress
                  </span>
                </a>
              </li>
            </ul>
          </div>
          <section
            className="wizard-pf-row"
          >
            <div
              className="wizard-pf-sidebar"
            >
              <ul
                className="list-group"
              >
                <li
                  className="active list-group-item"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    <span
                      className="wizard-pf-substep-number"
                    >
                      1A.
                    </span>
                    <span
                      className="wizard-pf-substep-title"
                    >
                      Details
                    </span>
                  </a>
                </li>
                <li
                  className="list-group-item"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    <span
                      className="wizard-pf-substep-number"
                    >
                      1B.
                    </span>
                    <span
                      className="wizard-pf-substep-title"
                    >
                      Settings
                    </span>
                  </a>
                </li>
              </ul>
              <ul
                className="hidden list-group"
              >
                <li
                  className="list-group-item"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    <span
                      className="wizard-pf-substep-number"
                    >
                      2A.
                    </span>
                    <span
                      className="wizard-pf-substep-title"
                    >
                      Details
                    </span>
                  </a>
                </li>
                <li
                  className="list-group-item"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    <span
                      className="wizard-pf-substep-number"
                    >
                      2B.
                    </span>
                    <span
                      className="wizard-pf-substep-title"
                    >
                      Settings
                    </span>
                  </a>
                </li>
              </ul>
              <ul
                className="hidden list-group"
              >
                <li
                  className="list-group-item"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    <span
                      className="wizard-pf-substep-number"
                    >
                      3A.
                    </span>
                    <span
                      className="wizard-pf-substep-title"
                    >
                      Summary
                    </span>
                  </a>
                </li>
                <li
                  className="list-group-item"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    <span
                      className="wizard-pf-substep-number"
                    >
                      3B.
                    </span>
                    <span
                      className="wizard-pf-substep-title"
                    >
                      Progress
                    </span>
                  </a>
                </li>
              </ul>
            </div>
            <div
              className="wizard-pf-main"
            >
              <div
                className="wizard-pf-contents"
              >
                <form
                  className="form-horizontal"
                >
                  <div
                    className="form-group required"
                  >
                    <label
                      className="col-sm-2 control-label"
                    >
                      Name
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <input
                        className="form-control"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      className="col-sm-2 control-label"
                    >
                      Description
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <textarea
                        className="form-control"
                        rows="2"
                      />
                    </div>
                  </div>
                </form>
              </div>
              <div
                className="wizard-pf-contents hidden"
              >
                <form
                  className="form-horizontal"
                >
                  <div
                    className="form-group required"
                  >
                    <label
                      className="col-sm-2 control-label"
                    >
                      Lorem ipsum
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <input
                        className="form-control"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      className="col-sm-2 control-label"
                    >
                      Dolor
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <textarea
                        className="form-control"
                        rows="2"
                      />
                    </div>
                  </div>
                </form>
              </div>
              <div
                className="wizard-pf-contents hidden"
              >
                <form
                  className="form-horizontal"
                >
                  <div
                    className="form-group required"
                  >
                    <label
                      className="col-sm-2 control-label"
                    >
                      Aliquam
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <input
                        className="form-control"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      className="col-sm-2 control-label"
                    >
                      Fermentum
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <textarea
                        className="form-control"
                        rows="2"
                      />
                    </div>
                  </div>
                </form>
              </div>
              <div
                className="wizard-pf-contents hidden"
              >
                <form
                  className="form-horizontal"
                >
                  <div
                    className="form-group required"
                  >
                    <label
                      className="col-sm-2 control-label"
                    >
                      Consectetur
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <input
                        className="form-control"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      className="col-sm-2 control-label"
                    >
                      Adipiscing
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <textarea
                        className="form-control"
                        rows="2"
                      />
                    </div>
                  </div>
                </form>
              </div>
              <div
                className="wizard-pf-contents hidden"
              >
                <div
                  className="wizard-pf-process blank-slate-pf"
                >
                  <div
                    className="spinner spinner-lg blank-slate-pf-icon"
                  />
                  <h3
                    className="blank-slate-pf-main-action"
                  >
                    Deployment in progress
                  </h3>
                  <p
                    className="blank-slate-pf-secondary-action"
                  >
                    Lorem ipsum dolor sit amet, porta at suspendisse ac, ut wisi vivamus, lorem sociosqu eget nunc amet.
                     
                  </p>
                </div>
              </div>
            </div>
          </section>
        </div>
        <div
          className="wizard-pf-footer"
        >
          <button
            className="btn-cancel btn btn-default"
            disabled={false}
            type="button"
          >
            Cancel
          </button>
          <button
            className="btn btn-default"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            <span
              className="i fa fa-angle-left"
            />
            Back
          </button>
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Next
            <span
              className="i fa fa-angle-right"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Wizard loading renders properly 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-12"
  >
    <div
      className="wizard-pf"
    >
      <div>
        <div
          className="wizard-pf-header"
        >
          <h4
            className="wizard-pf-title"
          >
            Wizard Title
          </h4>
        </div>
        <div
          className="wizard-pf-body"
        >
          <section
            className="wizard-pf-row"
          >
            <div
              className="wizard-pf-main"
            >
              <div
                className="blank-slate-pf"
              >
                <div
                  className="spinner spinner-lg blank-slate-pf-icon"
                />
                <h3
                  className="blank-slate-pf-main-action"
                >
                  Loading Wizard
                </h3>
                <p
                  className="blank-slate-pf-secondary-action"
                >
                  Lorem ipsum dolor sit amet, porta at suspendisse ac, ut wisi vivamus, lorem sociosqu eget nunc amet.
                   
                </p>
              </div>
            </div>
          </section>
        </div>
        <div
          className="wizard-pf-footer"
        >
          <button
            className="btn-cancel btn btn-default"
            disabled={false}
            type="button"
          >
            Cancel
          </button>
          <button
            className="btn btn-default"
            disabled={true}
            type="button"
          >
            <span
              className="i fa fa-angle-left"
            />
            Back
          </button>
          <button
            className="btn btn-primary"
            disabled={true}
            type="button"
          >
            Next
            <span
              className="i fa fa-angle-right"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
